compiler/damlc/lib/DA/Cli/Damlc/DependencyDb.hs

```diff
@@ -0,0 +1,204 @@
+-- Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Cli.Damlc.DependencyDb
+    ( installDependencies
+    , dependenciesDir
+    , queryDependencies
+    , queryDependencyDalfs
+    , queryDependencyMain
+    , queryDataDependencies
+    , mainDir
+    , dalfsDir
+    , configDir
+    , sourcesDir
+    ) where
+
+import qualified "zip-archive" Codec.Archive.Zip as ZipArchive
+import Control.Exception.Safe (tryAny)
+import Control.Monad
+import DA.Daml.Compiler.Dar
+import DA.Daml.Compiler.DecodeDar (DecodedDalf(..), decodeDalf)
+import DA.Daml.Compiler.ExtractDar (ExtractedDar(..), extractDar)
+import qualified DA.Daml.LF.Ast as LF
+import DA.Daml.Options.Types
+import DA.Daml.Package.Config
+import qualified DA.Pretty
+import Data.Aeson (eitherDecodeFileStrict', encode)
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BSL
+import Data.List.Extra
+import Data.Set (Set)
+import qualified Data.Set as Set
+import qualified Data.Text.Extended as T
+import Development.IDE.Types.Location
+import GHC.Fingerprint
+import "ghc-lib-parser" Module (unitIdString)
+import System.Directory.Extra
+import System.FilePath
+import System.IO.Extra
+
+-- Constants / Conventions
+--------------------------
+dependenciesDir :: Options -> NormalizedFilePath -> FilePath
+dependenciesDir opts projRoot =
+    fromNormalizedFilePath projRoot </> projectDependenciesDatabase </>
+    lfVersionString (optDamlLfVersion opts)
+
+fingerprintFile :: FilePath -> FilePath
+fingerprintFile depsDir = depsDir </> "fingerprint.json"
+
+mainDir, dalfsDir, configDir, sourcesDir, sdkVersionFile, depMarkerFile, dataDepMarkerFile :: FilePath -> FilePath
+mainDir depPath = depPath </> "main"
+dalfsDir depPath = depPath </> "dalfs"
+configDir depPath = depPath </> "config"
+sourcesDir depPath = depPath </> "sources"
+sdkVersionFile depPath = depPath </> "sdk-version"
+depMarkerFile depPath = depPath </> "_dependency_"
+dataDepMarkerFile depPath = depPath </> "_data_dependency_"
+
+
+
+-- Dependency installation
+--------------------------
+-- | Install all dependencies to the .daml/dependencies directory.
```

Comment:
Are the arguments filepaths to dars, dalfs, both? If so, let’s use `FilePath` instead of `String` and either way let’s document this.

Reply:
can be `FilePath` or the names of base-packages, or soon package ids. +1 on the documentation.

Reply:
Changed the type of data-dependencies to `FilePath` and added docs.